hip/base/config.hip.hpp

```diff
@@ -6,15 +6,13 @@
 #define GKO_HIP_BASE_CONFIG_HIP_HPP_
 
 
-#include <hip/hip_runtime.h>
-
-
 #include <hip/device_functions.h>
 
 
 #include <ginkgo/core/base/types.hpp>
 
 
+#include "common/cuda_hip/base/runtime.hpp"
```

Comment:
I think you shouldn't change the order of the runtime header. It causes compilation errors in some cases.

Reply:
That's a symptom of missing includes in our other files. Include order should not matter

Reply:
Is it really just because of our header? I thought you had to include `hip/hip_runtime.h` before other hip includes. I guess it could be worthwhile to investigate which of our header are causing trouble.

Reply:
the issue was hip/device_functions.h, which is not a self-sufficient header file and caused the compilation issues.

Reply:
Thanks for finding this. THat is a better solution than forcing the include order.